packaging/docker/Dockerfile

```diff
@@ -13,35 +21,42 @@ RUN mkdir crates
 RUN cargo new --lib crates/pedersen
 # Correct: --lib. We'll handle the binary later.
 RUN cargo new --lib crates/pathfinder
-COPY Cargo.toml Cargo.toml
-
-COPY crates/pathfinder/Cargo.toml crates/pathfinder/Cargo.toml
-COPY crates/pedersen/Cargo.toml crates/pedersen/Cargo.toml
-COPY crates/pedersen/benches crates/pedersen/benches
+RUN cp /tmp/pathfinder/Cargo.toml Cargo.toml
 
-RUN RUSTFLAGS='-L/usr/lib -Ctarget-feature=-crt-static' cargo build --release --target=x86_64-unknown-linux-musl
+RUN cp /tmp/pathfinder/crates/pathfinder/Cargo.toml crates/pathfinder/Cargo.toml
+RUN cp /tmp/pathfinder/crates/pedersen/Cargo.toml crates/pedersen/Cargo.toml
+RUN cp -R /tmp/pathfinder/crates/pedersen/benches crates/pedersen/benches
 
+RUN RUSTFLAGS='-L/usr/lib -Ctarget-feature=-crt-static' cargo build --release
 
 # Compile the actual libraries and binary now
-COPY . .
+RUN cp -R /tmp/pathfinder/* .
 
-# Mark these for re-compilation 
+# Mark these for re-compilation
 RUN touch crates/pathfinder/src/lib.rs
 RUN touch crates/pedersen/src/lib.rs
 
-RUN RUSTFLAGS='-L/usr/lib -Ctarget-feature=-crt-static' cargo build --release --target=x86_64-unknown-linux-musl --bin pathfinder
+RUN RUSTFLAGS='-L/usr/lib -Ctarget-feature=-crt-static' cargo build --release
+
+# Copy binary to hardcoded path so we can copy it to the `runner` image later on
+RUN mkdir bins && cp $(find ./target -name 'pathfinder') ./bins/pathfinder
```

Comment:
What do you mean? I added that cause rust tends to build bins to /target/<arch>, so I copy that to a pre-determined directory that I can later find it and copy it to the final image.

Reply:
Locally I removed `--target=x86_64-unknown-linux-musl` from the earlier `cargo build` calls and it put the binaries in the deterministic and predictable `/usr/src/pathfinder/target/release/` directory instead of `/usr/src/pathfinder/target/x86_64-unknown-linux-musl/release/`. 

By not specifying the target at build time, it should always use this directory no matter the arch, so we don't need to copy them to `./bins/pathfinder`.